core/trino-main/src/main/java/io/trino/operator/LookupJoinPageBuilder.java

```diff
@@ -122,6 +122,11 @@ else if (length == probeBlock.getPositionCount()) {
                 verify(probeIndices[length - 1] - probeIndices[0] == length - 1);
                 blocks[i] = probeBlock.getRegion(probeIndices[0], length);
             }
+
+            // unwrap loaded blocks
+            if (blocks[i].isLoaded()) {
+                blocks[i] = blocks[i].getLoadedBlock();
```

Comment:
`isLoaded` defaults to true, so this will be called for all basic blocks.  I'm not sure that matters... just pointing this out.

Reply:
I think it's better to do this check here instead of pushing lazy, but loaded blocks which would be checked for every position